devel/libecl/src/smspec_node.c

```diff
@@ -737,9 +737,11 @@ smspec_node_type * smspec_node_alloc_lgr( ecl_smspec_var_type var_type ,
 }
 
 smspec_node_type* smspec_node_alloc_copy( const smspec_node_type* node ) {
-    if( !node ) return NULL;
 
     smspec_node_type* copy = util_malloc( sizeof * copy );
+
+    if( !node ) return NULL;
```

Comment:
Thank you for the fix; I want to merge this - however the current fix will leak. I suggest instead:
```C
if (!node) return NULL;
{ 
     smspec_node_type * copy = util_malloc( sizeof * copy );
     ....
     ....
     return node;
}
}
```